src/lib/utils/tests/test_login.py

```
@@ -541,6 +697,34 @@ def test_oauth_flow_falls_back_to_config_client_id(self):
         self.assertEqual(mock_post.call_args.kwargs['data']['client_id'],
                          'config-client')
 
+    def test_oauth_flow_rejects_refresh_without_id_token(self):
+        expired_token = _make_jwt({'exp': int(time.time()) - 60})
+        storage = login.TokenLoginStorage(
+            id_token=expired_token,
+            refresh_token='old-refresh',
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "refresh_token"

(S106)